pandas/core/frame.py

```diff
@@ -8612,7 +8612,7 @@ def melt(
         ),
     )
     def diff(self, periods: int = 1, axis: Axis = 0) -> DataFrame:
-        if not isinstance(periods, int):
+        if not lib.is_integer(periods):
```

Comment:
```pandas/core/frame.py:8616: error: "int" has no attribute "is_integer"  [attr-defined]```

Reply:
that's pretty weird, looks like a mypy problem to me

Reply:
My bad, this actually refers to the line below ```if not (is_float(periods) and periods.is_integer()):```.